pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -146,8 +153,41 @@ func (r *TokenREST) Create(ctx context.Context, name string, obj runtime.Object,
 				return nil, errors.NewBadRequest(fmt.Sprintf("cannot bind token for serviceaccount %q to pod running with different serviceaccount name.", name))
 			}
 			uid = pod.UID
+			if utilfeature.DefaultFeatureGate.Enabled(features.ServiceAccountTokenPodNodeInfo) {
+				if nodeName := pod.Spec.NodeName; nodeName != "" {
+					newCtx := newContext(ctx, "nodes", nodeName, "", api.SchemeGroupVersion.WithKind("Node"))
+					// set ResourceVersion=0 to allow this to be read/served from the apiservers watch cache
+					nodeObj, err := r.nodes.Get(newCtx, nodeName, &metav1.GetOptions{ResourceVersion: "0"})
+					if err != nil {
+						nodeObj, err = r.nodes.Get(newCtx, nodeName, &metav1.GetOptions{}) // fallback to a live lookup on any error
+					}
+					switch {
+					case errors.IsNotFound(err):
```

Comment:
I've lost track of the layer we're in ... would this be a `k8s.io/apimachinery/pkg/api/errors#IsNotFound()` or a `k8s.io/apiserver/pkg/storage#IsNotFound()` error?

Reply:
(if there's a test for this case, that proves this condition is correct)

Reply:
We already have an integration test that ensures the node name is still set if the corresponding node object does not exist, which implies that we *are* correctly hitting this `IsNotFound` case :) https://github.com/kubernetes/kubernetes/pull/120780/files/b15580b6d7d8a494878d68459914c6da902aa9f1#diff-14c09ced561b5b87eda4a95c643d6f979cb3ec756f71dd385a85c956b1d1a31dR415